cpp/src/join/hash_join.cu

```diff
@@ -39,9 +39,9 @@ class row_contains_null {
  public:
   row_contains_null(bitmask_type const* row_bitmask) : _row_bitmask{row_bitmask} {}
 
-  __device__ __inline__ bool operator()(const pair_type& pair) const noexcept
+  __device__ __inline__ bool operator()(const size_type& i) const noexcept
   {
-    return _row_bitmask == nullptr or cudf::bit_is_set(_row_bitmask, pair.second);
+    return _row_bitmask == nullptr or cudf::bit_is_set(_row_bitmask, i);
```

Comment:
`_row_bitmask == nullptr` can never be true now, right? We'll only ever call `insert_if` when we've materialized a row bitmask.

Reply:
Yes, you are right.